glue/viewers/image/qt/tests/test_python_export.py

```diff
@@ -57,31 +87,70 @@ def test_simple_visual(self, tmpdir):
         self.viewer.state.layers[0].bias = 0.6
         self.assert_same(tmpdir)
 
-    def test_slice(self, tmpdir):
-        self.viewer.state.x_att = self.data.pixel_component_ids[1]
-        self.viewer.state.y_att = self.data.pixel_component_ids[0]
+    @pytest.mark.parametrize('coords', [None, 'wcs', 'affine'])
+    def test_slice(self, tmpdir, coords):
+        if coords is None:
+            data = self.data
+        else:
+            data = getattr(self, f'data_{coords}')
+            self.viewer.add_data(data)
+            self.viewer.remove_data(self.data)
```

Comment:
I wonder if this could perhaps be implemented as a pytest fixture to avoid the repetition in each test?

Reply:
@astrofrog I implemented this as a helper function now, but could not figure out yet how to use it as a fixture in light of https://docs.pytest.org/en/stable/deprecations.html#calling-fixtures-directly (or if that would actually be helpful).